engineioxide/src/engine.rs

```diff
@@ -83,7 +85,17 @@ impl<H: EngineIoHandler> EngineIo<H>
         self.handler.on_connect(&socket);
 
         let packet = OpenPacket::new(TransportType::Polling, sid, &self.config);
-        let packet: String = Packet::Open(packet).try_into()?;
+        #[allow(unused_mut)]
+        let mut packet: String = Packet::Open(packet).try_into()?;
+        cfg_if! {
+            if #[cfg(feature = "v3")] {
+                // The V3 protocol requires the packet length to be prepended to the packet.
+                // It doesn't use a packet separator like the V4 protocol (and up).
+                if protocol == ProtocolVersion::V3 {
+                    packet = format!("{}:{}", packet.chars().count(), packet);
+                }
+            }
+        }
```

Comment:
```suggestion
       let packet = {
            #[cfg(feature = "v3")]
            {
                let mut packet = packet;
                // The V3 protocol requires the packet length to be prepended to the packet.
                // It doesn't use a packet separator like the V4 protocol (and up).
                if protocol == ProtocolVersion::V3 {
                    packet = format!("{}:{}", packet.chars().count(), packet);
                }
                packet
            }
            #[cfg(not(feature = "v3"))]
            packet
        };

```

it will remove `#[allow(unused_mut)]`. And for me it will be as clear as before

Reply:
Nice, thank you! I'll apply your suggestion.